plugins/sf2_player/sf2_player.cpp

```diff
@@ -780,6 +784,13 @@ void sf2Instrument::deleteNotePluginData( NotePlayHandle * _n )
 									// do it here
 	{
 		noteOff( pluginData );
+		if( m_playingNotes.indexOf( _n ) >= 0 )
+		{
+			m_playingNotesMutex.lock();
+			m_playingNotes.remove( m_playingNotes.indexOf( _n ) );
```

Comment:
There's the possibility that `m_playingNotes.indexOf( _n ) >= 0` evaluates to true, then another thread removes `_n`, and then we obtain a lock and attempt to call `m_playingNotes.remove( m_playingNotes.indexOf( _n ) )`, which would be an error.

The simple solution is to move the locking and unlocking to just outside of the if statement.